classes/Cart.php

```diff
@@ -4083,7 +4083,8 @@ public function checkQuantities($returnProductOnFailure = false)
             if (
                 !$product['active'] ||
                 !$product['available_for_order'] ||
-                (!$product['allow_oosp'] && $product['stock_quantity'] < $product['cart_quantity'])
+                (!$product['allow_oosp'] && $product['stock_quantity'] < $product['cart_quantity']) &&
```

Comment:
@Codencode This won't work, we can't just check if the product is a pack. I think that we can just delete this line and keep just `!$product['active'] || !$product['available_for_order']`, because the quantity is checked on the line below. :-)

Reply:
@Hlavtox,

just to make sure I understand,
would you do it like this?
```
            if (
                !$product['active'] ||
                !$product['available_for_order']
            ) {
                return $returnProductOnFailure ? $product : false;
            }
```

I did a test and it's actually correct, it works

Reply:
Ok done.

What about that other comment? https://github.com/PrestaShop/PrestaShop/pull/35372#issuecomment-1956177017

I can proceed?